docs/porting/target/spi.md

```diff
@@ -0,0 +1,133 @@
+    - if `rx_len` > `tx_len` then it sends `(rx_len-tx_len)` additional `fill_symbol` to the bus.
+  - In Half-duplex mode :
+    - as master, `spi_transfer()` sends `tx_len` symbols and then reads `rx_len` symbols.
+    - as slave, `spi_transfer()` receives `rx_len` symbols and then sends `tx_len` symbols.
+- `spi_transter_async()` schedules a transfer to be process the same way `spi_transfer()` would have but asynchronously.
+- `spi_transter_async()` returns immediately with a boolean indicating whether the transfer was successfully scheduled or not.
+- The callback given to `spi_transfer_async()` is invoked when the transfer completes (with a success or an error).
+- `spi_transfer_async()` saves the handler and the `ctx` pointer.
+- The `ctx` is passed to the callback on transfer completion.
+- Unless if the transfer is aborted, the callback is invoked on completion. The completion maybe when all symbols have been transmitted
+  or when in slave mode the master de-asserts the chip select.
+- The `spi_transfer_async()` function may use the `DMAUsage` hint to select the appropriate async algorithm.
+- The `spi_async_event_t` must be filled with the number of symbol clocked on the bus during this transfer and a boolean value indicated if an error has occurred.
+- `spi_transfer_async_abort()` aborts an on-going async transfer.
+
+#### Undefined Behaviours
+
+- Calling `spi_init()` multiple times on the same `spi_t` without `spi_free()`'ing it first.
+- Calling any method other than `spi_init()` on a non-initialized or freed `spi_t`.
+- Passing both `miso` and `mosi` as `NC` to `spi_get_module` or `spi_init`.
+- Passing `miso` or `mosi` as `NC` on target that does not support half-duplex mode.
+- Passing `mclk` as `NC`  to `spi_get_module` or `spi_init`.
+- Passing an invalid pointer as `cap` to `spi_get_capabilities`.
+- Passing pins that cannot be on the same peripheral.
+- Passing an invalid pointer as `obj` to any method.
+- Giving a `ssel` pin to `spi_init()` when using in master mode.  
+  SS must be managed by hardware in slave mode and must **NOT** be managed by hardware in master mode.
+- Setting a frequency outside of the range given by `spi_get_capabilities()`.
+- Setting a frequency in slave mode.
+- Setting `bits` in `spi_format` to a value out of the range given by `spi_get_capabilities()`.
+- Passing an invalid pointer as `fill_symbol` to `spi_transfer` and `spi_transfer_async` while they would be required by the transfer (`rx_len != tx_len` or `tx==NULL`).
+- Passing an invalid pointer as `handler` to `spi_transfer_async`.
+- Calling `spi_transfer_async_abort()` while no async transfer is being processed (no transfer or a synchronous transfer).
+- In half-duplex mode, any mechanism (if any is present) to detect or prevent collision is implementation defined.
+
+#### Other requirements
+
+A target must also define these elements:
+
+- `#define SPI_COUNT (xxxxxU)`.
+  The number of SPI peripheral available on the device.
```

Comment:
Might be helpful to provide the filename where this should be added..

Reply:
I have declared in `PeripheralNames.h` in the reference implementation but is it THE best place ?
I looked sensible to me as the driver includes `hal/spi_api.h` that in turn includes `PeripheralNames.h`.

What do you think about it ?

Reply:
@ithinuel : Yes, I was referring to ```PeripheralNames.h``` . Can you maybe add a note in this line to add the number of peripherals there?